### PR TITLE
Fix: SVG and Json files caused errors in Drag from Desktop [ED-5529]

### PR DESCRIPTION
**Pull request by YitzhakAvm**

### CODE REVIEW

File: assets/dev/js/editor/document/elements/commands/reset-settings.js [hunk 1]
```diff
@@ -6,9 +6,9 @@ export class ResetSettings extends CommandHistory {
 	}
 
 	getHistory( args ) {
-		const { containers = [ args.container ] } = args;
+		const { containers = [ args.container ], silent = false } = args;
```
Comment (iNewLegend) [inline assets/dev/js/editor/document/elements/commands/reset-settings.js:9]:
```suggestion
		const { containers = [ args.container ], useHistory = false } = args;
```

File: assets/dev/js/editor/views/base-container.js [hunk 2]
```diff
@@ -93,25 +93,26 @@ module.exports = Marionette.CompositeView.extend( {
...
 		}
 
+		if ( elementor.helpers.maybeDisableWidget( model.widgetType ) ) {
+			return;
+		}
+
 		model = Object.assign( model, model.custom );
 
-		if ( 'section' === model.elType ) {
+		if ( 'document' !== container.type && 'section' === model.elType ) {
```
Comment (matipojo) [inline assets/dev/js/editor/views/base-container.js:108]: Explain

File: core/files/uploads-manager.php [hunk 3]
```diff
@@ -410,7 +410,7 @@ private function is_elementor_upload() {
 	 * @return bool
 	 */
 	private function is_elementor_media_upload() {
-		return isset( $_POST['uploadTypeCaller'] ) && 'elementor-media-upload' === $_POST['uploadTypeCaller']; // phpcs:ignore
+		return isset( $_REQUEST['uploadTypeCaller'] ) && 'elementor-media-upload' === $_REQUEST['uploadTypeCaller']; // phpcs:ignore
```
Comment (matipojo) [inline core/files/uploads-manager.php:413]: Explain why it may be sent via GET

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/document/elements/commands/reset-settings.js (hunk 1 above)
Comment (iNewLegend) [inline assets/dev/js/editor/document/elements/commands/reset-settings.js:9]:
```suggestion
		const { containers = [ args.container ], options = {} } = args,
				{ useHistory = false } = options;
```

File: assets/dev/js/editor/views/base-container.js (hunk 2 above)
Comment (ItzhakAvm) [reply assets/dev/js/editor/views/base-container.js:110]: Done.

File: core/files/uploads-manager.php (hunk 3 above)
Comment (ItzhakAvm) [reply core/files/uploads-manager.php:415]: Done.

File: assets/dev/js/editor/document/elements/commands/reset-settings.js (hunk 1 above)
Comment (ItzhakAvm) [reply assets/dev/js/editor/document/elements/commands/reset-settings.js]: Take a look.

File: assets/dev/js/editor/document/elements/commands-internal/reset-settings.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import CommandBase from 'elementor-api/modules/command-base';
+
+export class ResetSettings extends CommandBase {
+	validateArgs( args ) {
+		this.requireContainer( args );
+	}
+
+	apply( args ) {
+		$e.run( 'document/elements/reset-settings', { ...args, options: { useHistory: false } } );
```
Comment (iNewLegend) [inline assets/dev/js/editor/document/elements/commands-internal/reset-settings.js:9]: Internal command that runs public command, sound non-logical, which issue did you face?

Comment (ItzhakAvm) [reply assets/dev/js/editor/document/elements/commands-internal/reset-settings.js]: As discussed, I removed the internal command and used the regular one.

**Review by iNewLegend — APPROVED** [review]

File: assets/dev/js/editor/document/elements/commands/reset-settings.js [hunk 5]
```diff
@@ -6,20 +6,24 @@ export class ResetSettings extends CommandHistory {
 	}
 
 	getHistory( args ) {
-		const { containers = [ args.container ] } = args;
+		const { containers = [ args.container ], options: { useHistory = true } = {} } = args;
```
Comment (matipojo) [inline assets/dev/js/editor/document/elements/commands/reset-settings.js:9]: It should be a part of the `CommandHistory` not specific to this command/need.

File: assets/dev/js/editor/views/base-container.js [hunk 6]
```diff
@@ -93,25 +93,28 @@ module.exports = Marionette.CompositeView.extend( {
...
 		}
 
+		if ( elementor.helpers.maybeDisableWidget( model.widgetType ) ) {
+			return;
+		}
+
 		model = Object.assign( model, model.custom );
 
-		if ( 'section' === model.elType ) {
+		// When a section is about to be created inside an existing element, we want it to be an inner-section, unlike
```
Comment (matipojo) [inline assets/dev/js/editor/views/base-container.js:108]: Don't use `I/you/we` in documentation

File: assets/dev/js/editor/views/base-container.js [hunk 7]
```diff
@@ -93,25 +93,28 @@ module.exports = Marionette.CompositeView.extend( {
...
+		// a section that should be placed in the document level (for example, from a JSON template).
+		if ( 'document' !== container.type && 'section' === model.elType ) {
 			model.isInner = true;
 		}
 
-		if ( elementor.helpers.maybeDisableWidget( model.widgetType ) ) {
```
Comment (matipojo) [inline assets/dev/js/editor/views/base-container.js:106]: Why?

File: core/files/uploads-manager.php [hunk 8]
```diff
@@ -410,7 +410,9 @@ private function is_elementor_upload() {
 	 * @return bool
 	 */
 	private function is_elementor_media_upload() {
-		return isset( $_POST['uploadTypeCaller'] ) && 'elementor-media-upload' === $_POST['uploadTypeCaller']; // phpcs:ignore
+		// We use also use `uploadTypeCaller` as GET parameter when using the WP Media Library REST API, where the whole
```
Comment (matipojo) [inline core/files/uploads-manager.php:413]: ~We~

Comment (matipojo) [reply core/files/uploads-manager.php:413]:
```suggestion
		// We use `uploadTypeCaller` as GET parameter when using the WP Media Library REST API, where the whole
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/views/base-container.js (hunk 6 above)
Comment (ItzhakAvm) [reply assets/dev/js/editor/views/base-container.js]: Done.

File: core/files/uploads-manager.php (hunk 8 above)
Comment (ItzhakAvm) [reply core/files/uploads-manager.php]: Done.

File: assets/dev/js/editor/views/base-container.js (hunk 7 above)
Comment (ItzhakAvm) [reply assets/dev/js/editor/views/base-container.js:106]: I moved it upward.

File: assets/dev/js/editor/document/elements/commands/reset-settings.js (hunk 5 above)
Comment (ItzhakAvm) [reply assets/dev/js/editor/document/elements/commands/reset-settings.js:9]: We'll discuss this.

File: assets/dev/js/editor/views/base-container.js [hunk 9]
```diff
@@ -93,25 +93,28 @@ module.exports = Marionette.CompositeView.extend( {
...
 
+		if ( elementor.helpers.maybeDisableWidget( model.widgetType ) ) {
+			return;
+		}
+
 		model = Object.assign( model, model.custom );
 
-		if ( 'section' === model.elType ) {
+		// When a section is about to be created inside an existing element, it's converted to an inner-section, unless
+		// it's placed in the document level (for example, from a JSON template).
```
Comment (matipojo) [inline assets/dev/js/editor/views/base-container.js:109]: Try to use `isValidChild` otherwise add `container` and a `todo` comment

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/document/history/commands/base/base.js [hunk 10]
```diff
@@ -9,4 +9,29 @@ export default class Base extends CommandInternalBase {
...
+
+		if ( ! this.useHistory() ) {
+			throw new $e.modules.HookBreak;
+		}
+	}
+
+	onCatchApply( e ) {
+		if ( ! ( e instanceof $e.modules.HookBreak ) ) {
+			super.onCatchApply( e );
+		}
```
Comment (iNewLegend) [inline assets/dev/js/editor/document/history/commands/base/base.js:24]: Why?

Comment (ItzhakAvm) [reply assets/dev/js/editor/document/history/commands/base/base.js:24]: The common assumption of using both `start/end-log` commands and `CommandHistory` is that they're using this base class. If we check for the `useHistory` of the first currently running command here, we can determine whether to use history even in nested commands, and also whenever using `start/end-log` actions.

File: assets/dev/js/editor/views/base-container.js (hunk 9 above)
Comment (ItzhakAvm) [reply assets/dev/js/editor/views/base-container.js]: Done.

File: assets/dev/js/editor/document/history/commands/base/base.js (hunk 10 above)
Comment (iNewLegend) [reply assets/dev/js/editor/document/history/commands/base/base.js:24]: I think that everyone that manually handle the _History_  should somehow know what the status of history, the infa that should remain the same... the only code that should stay in `CommandHistory` is the one that check first args trace in the `consturctor`.

Comment (ItzhakAvm) [reply assets/dev/js/editor/document/history/commands/base/base.js:24]: Moved to `CommandHistory`.

Comment (iNewLegend) [reply assets/dev/js/editor/document/history/commands/base/base.js:24]: ?

File: assets/dev/js/editor/document/commands/base/command-history.js [hunk 11]
```diff
@@ -42,7 +42,8 @@ export default class CommandHistory extends CommandBase {
 	 * @returns {boolean}
 	 */
 	isHistoryActive() {
-		return elementor.documents.getCurrent().history.getActive();
+		return elementor.documents.getCurrent().history.getActive() &&
+			false !== $e.commands.getCurrentFirstTraceArgs().options?.useHistory;
```
Comment (iNewLegend) [inline assets/dev/js/editor/document/commands/base/command-history.js:46]: Would be great if you can add test in `CommandHistory.spec` that covers `useHistory`

**Review by iNewLegend — APPROVED** [review]

**Review by matipojo — CHANGES_REQUESTED** [review]
Please add a QUnit test for the `useHistory`.
Please add a Playwright test for the dragigng.

File: tests/qunit/tests/assets/dev/js/editor/document/commands/base/command-history.spec.js [hunk 12]
```diff
@@ -13,6 +14,32 @@ jQuery( () => {
 					new Error( 'CommandHistory.getHistory() should be implemented, please provide \'getHistory\' functionality.' )
 				);
 			} );
+
+			QUnit.only( 'isHistoryActive(): `useHistory` is `false`', async ( assert ) => {
```
Comment (matipojo) [inline tests/qunit/tests/assets/dev/js/editor/document/commands/base/command-history.spec.js:18]:
```suggestion
			QUnit.test( 'isHistoryActive(): `useHistory` is `false`', async ( assert ) => {
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/components/browser-import/session.js [hunk 13]
```diff
@@ -89,14 +89,18 @@ export default class Session {
 	 * @returns {*}
 	 */
 	resolve( containers ) {
+		if ( Object.values( containers ).some( ( element ) => 'section' === element.model.get( 'elType' ) ) ) {
```
Comment (matipojo) [inline assets/dev/js/editor/components/browser-import/session.js:92]: Can you scroll to the first new imported element (on after the import is completed)

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/common/assets/js/api/components/wordpress/commands-data/media.js [hunk 14]
```diff
@@ -50,12 +55,16 @@ export class Media extends CommandData {
...
+		if (
 			! window._wpPluploadSettings.defaults.filters.mime_types[ 0 ].extensions.split( ',' ).includes(
 				this.file.name.split( '.' ).pop()
-			)
+			) &&
+			! elementor.config.filesUpload.unfilteredFiles
 		) {
-			throw new Error( __( 'Sorry, this file type is not permitted for security reasons.', 'elementor' ) );
+			FilesUploadHandler.getUnfilteredFilesNotEnabledDialog( () => {} ).show();
+			return;
```
Comment (matipojo) [inline core/common/assets/js/api/components/wordpress/commands-data/media.js:67]: On cancel from the SVG dialog, the panel still shows the dragged SVG

**Review by matipojo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/components/browser-import/session.js (hunk 13 above)
Comment (ItzhakAvm) [reply assets/dev/js/editor/components/browser-import/session.js:92]: Done.

File: core/common/assets/js/api/components/wordpress/commands-data/media.js (hunk 14 above)
Comment (ItzhakAvm) [reply core/common/assets/js/api/components/wordpress/commands-data/media.js:67]: Done.

**Pull request merged**